sys/shell/shell.c

```diff
@@ -44,6 +44,12 @@ static void _putchar(int c) {
 #endif
 #endif
 
+#ifdef MODULE_NEWLIB
```

Comment:
Because we don't need to flush after _every_ character. I do think, however, that it would be better if stdio was in unbuffered mode: it fits better with the HW.